src/TicketingSystem.PurchasesAPI/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,186 @@
+    public class OrdersController : ControllerBase
+    {
+        private readonly IEventService _eventService;
+        private readonly IEventSeatService _eventSeatService;
+        private readonly IPaymentService _paymentService;
+        private readonly ITicketService _ticketService;
+        private readonly ICartItemService _cartItemService;
+        private readonly IMapper _mapper;
+
+        public OrdersController(
+            ICartItemService cartItemService,
+            ITicketService ticketService,
+            IPaymentService paymentService,
+            IEventService eventService,
+            IEventSeatService eventSeatService,
+            IMapper mapper)
+        {
+            _cartItemService = cartItemService;
+            _ticketService = ticketService;
+            _paymentService = paymentService;
+            _eventService = eventService;
+            _eventSeatService = eventSeatService;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Returns a list a list of items in a cart
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("carts/{cartId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetCartItems([FromRoute] string cartId)
+        {
+            var items = await _cartItemService.FilterAsync(ci => ci.CartId == cartId);
+
+            return Ok(items);
+        }
+
+        /// <summary>
+        /// Adds a seat to the cart
+        /// </summary>
+        /// <returns>A cart state (with total amount)</returns>
+        [HttpPost]
+        [Route("carts/{cartId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> AddSeatToCart([FromRoute] string cartId, [FromBody] AddCartModel model)
+        {
+            var existingEvent = await _eventService.GetByIdAsync(model.EventId);
+            if(existingEvent is null)
+            {
+                return NotFound(nameof(model.EventId));
+            }
+
+            var existingSeatId = await _eventSeatService.GetByIdAsync(model.SeatId);
+            if (existingSeatId is null)
+            {
+                return NotFound(nameof(model.SeatId));
+            }
+
+            return Ok(await AddSeatToCart(model, cartId));
+        }
+
+        /// <summary>
+        /// Adds a seat to the cart. Returns Aacart state (with total amount)
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete]
+        [Route("carts/{cartId}/events/{eventId}/seats/{seatId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> AddSeatToCart([FromRoute] string cartId, [FromRoute] string eventId, [FromRoute] string seatId)
+        {
+            var existingEvent = await _eventService.GetByIdAsync(eventId);
```

Comment:
AddSeatsToCart -> should be Remove or Delete? Please, check other endpoints for naming as well.

Reply:
Checked & fixed